ui/src/components/ThemeMode.tsx

```diff
@@ -3,44 +3,53 @@
 // SPDX-License-Identifier: GPL-v3.0-or-later WITH GPL-3.0-linking-exception
 
 import { useColorScheme } from "@mui/material";
-import React, { createContext } from "react";
+import React, { createContext, useCallback } from "react";
+
+export enum CurrentMode {
+  Auto = 0,
+  Light = 1,
+  Dark = 2
+}
 
 interface ThemeProps {
   changeTheme: () => void,
-  currentTheme?: 'dark' | 'light',
-  getTheme: 0 | 1 | 2,
-  systemColor: 'dark' | 'light'
+  currentTheme: 'dark' | 'light',
+  choosenTheme: CurrentMode
 }
 
 export const ThemeContext = createContext<ThemeProps>({
   currentTheme: window.matchMedia("(prefers-color-scheme: dark)").matches ? 'dark' : 'light',
```

Comment:
[This comment](https://github.com/vitest-dev/vitest/issues/821#issuecomment-1046954558) looks promising for fixing this `TypeError` from `vitest`.

Reply:
I see.